### ISSUE TITLE
Features: OpenID Connect Auth

### ISSUE THREAD

**Issue by ptman** [nocodb/nocodb#742]
OpenID Connect isn't a long step from Google Auth, but would add support for e.g. Azure AD, Github, Gitlab, KeyCloak, ...

**Comment by stregouet** [comment]
Hi,
I also need this feature! so I'm trying to implement this with [passport-openidconnect](https://www.npmjs.com/package/passport-openidconnect) but I'm stuck with error on missing express session (see `Error` instanciation in [passport-openidconnect repo](https://github.com/jaredhanson/passport-openidconnect/blob/master/lib/state/session.js#L39)).
Any idea how I can access a session in nocodb?

**Comment by NexZhu** [comment]
@stregouet Hi, I'm curious have you made any progress on this?

**Comment by stregouet** [comment]
nope, I'm still stuck with this missing session middleware.

**Comment by syji35** [comment]
I think express-session middleware is missing.

Check stackoverflow post [here](https://stackoverflow.com/questions/22298033/nodejs-passport-error-oauthstrategy-requires-session-support)

**Comment by syji35** [comment]
Hi,
I tried to implement with 'openid-client' by following the following [article](https://codeburst.io/how-to-implement-openid-authentication-with-openid-client-and-passport-in-node-js-43d020121e87).  

I have the same problem as on #517 :  the application does not start anymore. 

> nocodb@0.11.29 watch:run
> nodemon -e ts,js -w ./src -x "ts-node src/example/docker --log-error --project tsconfig.json"

```
[nodemon] 2.0.7
[nodemon] to restart at any time, enter `rs`
[nodemon] watching path(s): src\**\*
[nodemon] watching extensions: ts,js
[nodemon] starting `ts-node src/example/docker --log-error --project tsconfig.json`
```

I am also stuck.

**Comment by c-nv-s** [comment]
this feature would be an amazing help for me

**Comment by helmut72** [comment]
+1 OIDC is easier than SAML and LDAP can't do single sign on.

**Comment by almereyda** [comment]
There is a slight chance at implementing an OpenID Connect adapter at:

- https://github.com/nocodb/nocodb/pull/3979

**Comment by timmwille** [comment]
> There is a slight chance at implementing an OpenID Connect adapter at:
> 
>     * [feat: OpenID Connect authentication #3979](https://github.com/nocodb/nocodb/pull/3979)

I guess this is still under analysis - but some statement from the core devs would be great to be able to evaluate if this is going into the more open for communities and networks direction or not to be able to choose an alternative early on or be patient and wait or go and fund/develop a working patch? 

What is the status on this? shall we wait? help? fund? skip? I'm also happy on having a call for this, to clarify

**Comment by c-nv-s** [comment]
been patiently waiting on the outcome of this feature and pull request https://github.com/nocodb/nocodb/pull/3979
because it is definitely going to determine whether I pursue with using this project or whether it gets dropped.
Especially if it turns out that it is not included for political reasons, because I'm tired of big companies like Redhat buying these types of projects, essentially claiming they support the open source community, when later on down the line we all discover that is not the case.

**Comment by Monsterlin2018** [comment]
NocoDB is a design I like very much. It is lightweight and focused, and can liberate my productivity at the basic data operation level.
It is impossible for NocoDB to take over all business needs. If nocodb doesn't have SSO/SAML, I can't integrate it with my other tools. Only NocoDB can be abandoned.
---
NocoDB是我非常喜欢的设计，轻巧、专注，能够在基本的数据操作层面上解放我的生产力。
NocoDB不可能接管所有的商业需求。如果nocodb没有 SSO/SAML，我无法将它集成到我的其他工具。被放弃的只能是NocoDB。

**Comment by ludat** [comment]
I've just spent like 3 hours implementing oidc integration and now I discover https://github.com/nocodb/nocodb/pull/3979. it's very frustrating.

Why is this issue still open? if you won't accept PRs you should either close this PR or make it more clear.

**Comment by timmwille** [comment]
> I've just spent like 3 hours implementing oidc integration (it kind of works) and now I discover #3979. it's very frustrating.
> 
> Why is this issue still open? if you won't accept PRs you should either close this PR or make it more clear.

Community spirit at nocodb might not be the main focus right now

**Comment by punkyard** [comment]
i agree, SSI is the base of personal and simple work, it's not pro, it's just a need in 2024 for an app to be integrated into individuals workflow ✌️

**Comment by sqkkyzx** [comment]
I must honestly say that I do use NocoDB for collaboration with my colleagues, providing them with a good interface to view internal database data. Due to budget constraints, I will not purchase an enterprise license and will only use the community version, even though SSO is a feature I urgently need. ChatGPT tells me that under the AGPL-3.0 license, since I have not modified or redistributed the code, and there is already a link to the source code repository in NocoDB's user interface, I can use it.

The contributors to NocoDB are indeed generous, providing many features in the community version, making them respectable pioneers. For example, there are no restrictions on configuring all colors for options, using all column types, connecting to all supported external databases, and continuously fixing bugs. These are the reasons I cannot leave NocoDB at the moment. Even without SSO, I only opened NocoDB accounts for 5 people in my workgroup, and I can manage manually.

However, although the following words are self-deprecating, I still want to tell the project team: Workers in small teams like me are actually like vultures, not particularly loyal to any project. We look for cost-saving software everywhere and will leave it for the next one when we find a better option. Our data volume is not as difficult to migrate as that of large enterprises. As far as I know, many similar projects are under development. Frankly speaking, while using NocoDB, I am also keeping an eye on their development progress. Of course, NocoDB is currently my favorite and the one that best meets my needs.

I don't want to repeat the open-source spirit, but I want to talk from a commercial perspective. As a small team, we seem insignificant for those real commercial activities and are likely to lack the budget to purchase software licenses. The traditional commercial license sales model makes me hesitant (especially referring to the non-transparent public prices, which require me to communicate with salespeople through online meetings or emails). If you indeed wish to provide additional features to generate revenue for the project, please be sure to note this: Listing each functional component with individual prices is easier for me to pay for than bundling them into an enterprise license. Additionally, as the team grows, I might eventually get some budget, but by then, my needs may also change. Just last week, I turned down a $4,000 no-code platform sale that allowed me to install 5 commercial plugins on the community version. I was tempted, but NocoDB is sufficient for now, and I don't need to spend the extra $4,000. However, next year my team will have more people, which is one of the challenges these projects face in commercialization. Spreadsheet applications are very useful, can be quickly launched, are truly no-code, and have sufficient functionality. But as the team size changes, more powerful software may be needed.

**Comment by o1lab** [comment]
Available in enterprise edition.
